cli/commands/cloud/cloud.go

```diff
@@ -197,6 +197,8 @@ func NewUpdateCloudServiceCommand() *cobra.Command {
 			res, err = client.UpdateCloudService(context.Background(), &orchestrator.UpdateCloudServiceRequest{
 				ServiceId: viper.GetString("id"),
 				Service: &orchestrator.CloudService{
+					// TODO(lebogg): Remove again if we decide to keep service ID in separate field above
+					Id:          viper.GetString("id"),
```

Comment:
Yea, maybe it should not part of this PR but it didn't and still doesn't feel right to use the ID in the way you say. Thus, you have to put the same ID twice (in REQ and in server object). Otherwise, maybe the DB overrides even the ID in the DB which should not be possible. Of course, we could mitigate this by adding a validation check that IDs should be the same or by implementing the DB Update operation accordingly (but see also comment in code with the differences of `gorm.Update()` vs `gorm.Save()`)

See also my comment above: "In UpdateXXX RPCs, should we add ID as separate field as in UpdateCloudService. Is this some convention? For the REST case we can add it via option anyway - I would omit it."

Reply:
You only need to put the ID in the request, not in the embedded service object. The reasoning behind this is that basically the update requests contains a list of key/values what to change in the original server object. Now instead of supply a extra message that contains all the fields as the original service object (with exception) of the id, it is convention to just use the service object itself and keep the ID empty.

Reply:
See our referenced style guide for examples https://cloud.google.com/apis/design/standard_methods#create.

Annoyingly, they are using the extra ID field for create, but omit it for update. However, they are using patch instead of put for updates.